apollo-router/src/plugins/rhai/engine.rs

```diff
@@ -218,6 +219,34 @@ mod router_method {
     }
 }
 
+#[export_module]
+mod status_code {
+    pub(crate) type StatusCode = http::StatusCode;
+
+    #[rhai_fn(name = "to_string", pure)]
+    pub(crate) fn status_code_to_string(status_code: &mut StatusCode) -> String {
+        status_code.as_str().to_string()
+    }
+
+    #[rhai_fn(name = "to_number", pure)]
+    pub(crate) fn status_code_to_number(status_code: &mut StatusCode) -> i64 {
+        status_code.as_u16() as i64
+    }
```

Comment:
`Rhai` has built-in functionality for converting between `String` and `Int`, so you could remove this and just have:

```
let s_as_num = parse_int(code.to_string())
```